app/src/main/java/com/example/lhi06/residue/FindAlbumActivity.java

```diff
@@ -1,36 +1,65 @@
 package com.example.lhi06.residue;
 
-import android.support.annotation.NonNull;
-import android.support.v7.app.AppCompatActivity;
+import android.content.Context;
 import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.widget.EditText;
+import android.widget.Toast;
 
 import com.squareup.picasso.Picasso;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.io.IOException;
 
 import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class FindAlbumActivity extends AppCompatActivity {
 
-    AlbumAdapter adapter;
+    private AlbumAdapter adapter;
+    private ItunesService service;
 
-    private List<Album> listAlbum = new ArrayList<>();
 
     @BindView(R.id.recycleView) RecyclerView recyclerView;
+    @BindView(R.id.request) EditText editText;
 
     @Override
-    protected void onCreate(@NonNull final Bundle savedInstanceState) {
+    protected void onCreate(@Nullable final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_find_album);
+        ButterKnife.bind(this);
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         adapter = new AlbumAdapter(Picasso.with(this));
-        // populateRecyclerViewValues():
+        recyclerView.setAdapter(adapter);
+        service = new ItunesService();
+
     }
 
-    private void populateRecyclerViewValues() {
-        // TODO: API buidler and Album list Populater
+    @OnClick(R.id.search_button)
+    public void onSearchClicked() {
+        try {
+            service.getArtistsAlbums(editText.getText().toString()).enqueue(new Callback<AlbumSearchResponse>() {
```

Comment:
Imagine you had another activity where you wanted to perform the same search. You would have to repeat the code to make it run in another thread. I would rather encapsulate that logic (running in another thread) in the service